File: Practica 1/P1/P1/EjerciciosJava/src/Ejercicios/Parcela.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Ejercicios;

/**
 *
 * @author antonio
 */
public class Parcela {

    private String nombre;
    private float precioCompra, precioEdificar, precioAlquilerBase;
    private int numCasas, numHoteles;
    private static float FACTORALQUILERCALLE = 1.0f;
    private static float FACTORALQUILERCASA = 1.0f;
    private static float FACTORALQUILERHOTEL = 4.0f;
    
    Parcela (String nombre, float precioCompra, float precioEdificar, float precioBaseAlquiler){
        
        this.nombre = nombre;
        this.precioCompra = precioCompra;
        this.precioEdificar = precioEdificar;
        this.precioAlquilerBase = precioBaseAlquiler;
        this.numCasas = 0;
        this.numHoteles = 0;
    }

    String getNombre() {
        return nombre;
    }

    float getPrecioCompra() {
        return precioCompra;
    }

    float getPrecioEdificar() {
        return precioEdificar;
    }

    int getNumCasas() {
        return numCasas;
    }

    int getNumHoteles() {
        return numHoteles;
    }
    
    float getPrecioAlquilerCompleto(){    
        return (precioAlquilerBase * FACTORALQUILERCALLE) * (FACTORALQUILERCASA + numCasas + numHoteles * FACTORALQUILERHOTEL);
    }
    
    boolean construirCasa(){
        numCasas+=1;
        return true;
    }
    
    boolean construirHotel(){
        numHoteles+=1;
        return true;
    }
    
    boolean igualdadIdentidad(Parcela otraParcela){
        boolean resultado;
        
        if(this == otraParcela)
            resultado = true;
        else
            resultado = false;
        
        return resultado;
    }
    
    boolean igualdadEstado(Parcela otraParcela){
        boolean resultado;
        
        resultado = this.nombre.equals(otraParcela.nombre) && this.precioCompra == otraParcela.precioCompra
                && this.precioEdificar == otraParcela.precioEdificar && this.precioAlquilerBase == otraParcela.precioAlquilerBase
                && this.numCasas == otraParcela.numCasas && this.numHoteles == otraParcela.numHoteles;
        
        return resultado;
    }
    
   
}

File: Practica 1/P1/P1/parcela.rb
#encoding UTF-8

class Parcela

  #@@factoral_alquiler_calle = 1.0
  #@@factoral_alquiler_casa = 1.0
  #@@factoral_alquiler_hotel = 4.0

  @factoral_alquiler_calle = 1.0
  @factoral_alquiler_casa = 1.0
  @factoral_alquiler_hotel = 4.0

  def initialize(nombre, precio_compra, precio_edificar, precio_alquiler_base)
    @nombre = nombre
    @precio_compra = precio_compra
    @precio_edificar = precio_edificar
    @precio_alquiler_base = precio_alquiler_base
    @num_casas = 0
    @num_hoteles = 0
  end

  def nombre
    @nombre
  end

  def self.factoral_alquiler_calle
    @factoral_alquiler_calle
  end

  def self.factoral_alquiler_casa
    @factoral_alquiler_casa
  end

  def self.factoral_alquiler_hotel
    @factoral_alquiler_hotel
  end

  def precio_compra
    @precio_compra
  end

  def precio_edificar
    @precio_edificar
  end

  def num_casas
    @num_casas
  end

  def num_hoteles
    @num_hoteles
  end

  def get_precio_alquiler_completo
    #(@precio_alquiler_base * @@factoral_alquiler_calle) * (@@factoral_alquiler_casa + @num_casas + @num_hoteles * @@factoral_alquiler_hotel)
    factoral_alquiler_casa = Parcela.factoral_alquiler_casa
    factoral_alquiler_calle = Parcela.factoral_alquiler_calle
    factoral_alquiler_hotel = Parcela.factoral_alquiler_hotel

    (@precio_alquiler_base * factoral_alquiler_calle) * (factoral_alquiler_casa + @num_casas + @num_hoteles * factoral_alquiler_hotel)
  end

  def construir_casa
    @num_casas+=1
    true
  end

  def construir_hotel
    @num_hoteles+=1
    true
  end

end

p1 = Parcela.new("Parcela1", 1000, 200, 100)
p2 = Parcela.new("Parcela2", 2000, 500, 350)

p1.construir_casa
p1.construir_casa

puts 'El precio de alquiler completo de la parcela 1 es ' + p1.get_precio_alquiler_completo.to_s

p2.construir_hotel

puts 'El precio de alquiler completo de la parcela 2 es ' + p2.get_precio_alquiler_completo.to_s